### PR TITLE
change mender-inventory-hostinfo to use the output of hostname

### PR DESCRIPTION
**Pull request by MuchoLucho**
Signed-off-by: Luis Ramirez Vargas <luis.ramirez@northern.tech>

# External Contributor Checklist

🚨 Please review the [guidelines for contributing](https://github.com/mendersoftware/mender/blob/master/CONTRIBUTING.md) to this repository.

- [x] Make sure that all commits have a [`Changelog`](https://github.com/mendersoftware/mender/blob/master/CONTRIBUTING.md#changelog-tags) tag. If nothing should be added to the Changelog, add a `Changelog: None` tag. If there is a change, add `Changelog: Commit`, or `Changelog: Title`, depending on what should be included in the changelog.

- [x] Make sure that all commits are signed with [`git --signoff`](https://git-scm.com/book/en/v2/Git-Tools-Signing-Your-Work). Also note that the signoff author must match the author of the commit.

### Description

By using `cat` you will print the comments in the file, while the actual command ignore them, if a user has in `/etc/hostname`:

```bash
#Any comment
actualHostName
```

The script will fail. Even busybox has the `hostname` command available
 
Thank you!

### CODE REVIEW

File: support/mender-inventory-hostinfo [hunk 1]
```diff
@@ -17,5 +17,5 @@ cat /proc/meminfo | awk '
 /MemTotal/ {printf("mem_total_kB=%d\n", $2)}
 '
 
-echo "hostname=$(cat /etc/hostname)"
+echo "hostname=$(hostname)"
```
Comment (merlin-northern) [inline support/mender-inventory-hostinfo:20]: maybe lets check if we have `hostname` command, if we do, then use it, if not, then fall back to `/etc/hostname`, and if that fails, we can even grep in the /etc/hosts for the first word in a first uncommented line for 127.0.0.1, and if that fails we fall back to `localhost` ? just a nit pick.

Comment (merlin-northern) [reply support/mender-inventory-hostinfo:20]: sorry for late answer, I only got ot it now. perhaps:
```bash
hostname="localhost"
hostname >/dev/null 2>&1 && hostname="$(hostname)"
[ "$hostname" == "" ] && [ -f /etc/hostname ] && hostname="cat /etc/hostname 2>/dev/null"
hostname=${hostname:-"localhost"}
```

File: support/mender-inventory-hostinfo [hunk 2]
```diff
@@ -17,5 +17,7 @@ cat /proc/meminfo | awk '
 /MemTotal/ {printf("mem_total_kB=%d\n", $2)}
 '
 
-echo "hostname=$(cat /etc/hostname)"
-
+hostname="localhost"
+hostname >/dev/null 2>&1 && hostname="$(hostname)"
+[ "$hostname" == "" ] && [ -f /etc/hostname ] && hostname="cat /etc/hostname 2>/dev/null"
```
Comment (merlin-northern) [inline support/mender-inventory-hostinfo:22]: small misspell with the double quotes, the above has to read:

```
[ "$hostname" == "" ] && [ -f /etc/hostname ] && hostname=$(cat /etc/hostname 2>/dev/null)
```

**Review by merlin-northern — CHANGES_REQUESTED** [review]

File: support/mender-inventory-hostinfo (hunk 2 above)
Comment (estape11) [reply support/mender-inventory-hostinfo]: Changes are ready

**Review by merlin-northern — APPROVED** [review]
lgtm, thank you. one minor comment: with this change it is possible that we send fully qualified hostnames, if we fall to the `hostname` command, in order to send only the hostname we could use `hostname -s`, but I checked, and it is ok to send the fully qualified hostname.

**Comment by merlin-northern** [comment]
@estape11 the commit message should look like:

```
fix: <nice description>

Changelog: <nice change log>
Ticket: None
Signed-off-by: you
```

and we need the licence fix.

File: support/mender-inventory-hostinfo [hunk 3]
```diff
@@ -17,5 +17,7 @@ cat /proc/meminfo | awk '
 /MemTotal/ {printf("mem_total_kB=%d\n", $2)}
 '
 
-echo "hostname=$(cat /etc/hostname)"
-
+hostname="localhost"
+hostname >/dev/null 2>&1 && hostname="$(hostname)"
+[ "$hostname" == "" ] && [ -f /etc/hostname ] && hostname=$(cat /etc/hostname 2>/dev/null)
```
Comment (merlin-northern) [inline support/mender-inventory-hostinfo:22]: @estape11 one thing I missed: it has to be a single `=`. this line has to look like that:

```
[ "$hostname" = "" ] && [ -f /etc/hostname ] && hostname=$(cat /etc/hostname 2>/dev/null)
```

**Review by merlin-northern — CHANGES_REQUESTED** [review]

File: support/mender-inventory-hostinfo (hunk 3 above)
Comment (estape11) [reply support/mender-inventory-hostinfo]: @merlin-northern ready to rock (in theory) 😊

**Review by merlin-northern — APPROVED** [review]
lgtm! (sorry for the annoying review)

**Comment by estape11** [comment]
> lgtm! (sorry for the annoying review)

Haha no worries, thank you for taking the time to review it, now whats next?

**Comment by merlin-northern** [comment]
> > lgtm! (sorry for the annoying review)
> 
> Haha no worries, thank you for taking the time to review it, now whats next?

you need to fix the commits (see the failing pipeline) and the license.

**Comment by lluiscampos** [comment]
@mender-test-bot start pipeline :ninja:

**Comment by lluiscampos** [comment]
Last force-push amended commit message.

Full pipeline [here](https://gitlab.com/Northern.tech/Mender/mender-qa/-/pipelines/704950132).

**Pull request merged**